### PR TITLE
[wait start 9.3+2.0] Enhance rule mactched logs with display in item. We add / display the…

### PR DESCRIPTION
**Pull request by ddurieux**
… criteria used in the rules

criteria_matchedrules

### CODE REVIEW

**Comment by Wolvverine** [comment]
PHP Notice:  Undefined index: PROCESSNUMBER 
https://travis-ci.com/Wolvverine/fusioninventory-for-glpi/jobs/154450170

**Comment by Wolvverine** [comment]
@ddurieux When is the release of version 9.3 + 2.0 planned?

**Comment by ddurieux** [comment]
I don't know yet, I have a little code  + add tests in these PR. so perhaps middle december, perhaps end od december (in my holidays :p)

File: inc/communicationnetworkinventory.class.php [hunk 1]
```diff
@@ -490,7 +492,8 @@ function rulepassed($items_id, $itemtype) {
             }
             $inputrulelog['items_id'] = $items_id;
             $inputrulelog['itemtype'] = $itemtype;
-            $inputrulelog['method'] = 'snmpinventory';
+            $inputrulelog['method'] = 'networkinventory';
+            $inputrulelog['criteria'] = $dbu->exportArrayToDB(unserialize($_SESSION['plugin_fusinvsnmp_datacriteria']));
```
Comment (trasher) [inline inc/communicationnetworkinventory.class.php]: You should just use `epoxrtArraytoDB` without any DbUtils instanciation

File: inc/inventoryruleimport.class.php [hunk 2]
```diff
@@ -764,6 +765,11 @@ function executeActions($output, $params, array $input = []) {
                            if (!isset($params['return'])) {
                               $inputrulelog['items_id'] = $items_id;
                               $inputrulelog['itemtype'] = $itemtype;
+                              $critinput = $this->criteriaInput;
+                              if (isset($critinput['class'])) {
+                                 unset($critinput['class']);
+                              }
+                              $inputrulelog['criteria'] = $dbu->exportArrayToDB($critinput);
```
Comment (trasher) [inline inc/inventoryruleimport.class.php]: same

File: inc/rulematchedlog.class.php [hunk 3]
```diff
@@ -300,6 +306,22 @@ function showForm($items_id, $itemtype) {
             }
          }
          echo "</td>";
+         echo "<td>";
+         $criteria = $dbu->importArrayFromDB($data['criteria']);
```
Comment (trasher) [inline inc/rulematchedlog.class.php]: same

File: inc/rulematchedlog.class.php [hunk 4]
```diff
@@ -300,6 +306,22 @@ function showForm($items_id, $itemtype) {
...
+            foreach ($criteria as $key=>$value) {
+               echo "<li>".$key.": ";
+               if (is_array($value)) {
+                  echo implode("<br>", $value);
+               } else {
+                  echo $value;
+               }
+               echo "</li>";
+            }
+            echo "</ul><hr style='border: 0;height: 1px;background-image: -webkit-linear-gradient(left, #f0f0f0, #8c8b8b, #f0f0f0);background-image: -moz-linear-gradient(left, #f0f0f0, #8c8b8b, #f0f0f0);background-image: -ms-linear-gradient(left, #f0f0f0, #8c8b8b, #f0f0f0);background-image: -o-linear-gradient(left, #f0f0f0, #8c8b8b, #f0f0f0);'>";
```
Comment (trasher) [inline inc/rulematchedlog.class.php]: Putting this in a CSS file would be better.

File: inc/communicationnetworkinventory.class.php (hunk 1 above)
Comment (ddurieux) [reply inc/communicationnetworkinventory.class.php]: Fixed

File: inc/inventoryruleimport.class.php (hunk 2 above)
Comment (ddurieux) [reply inc/inventoryruleimport.class.php]: Fixed

File: inc/rulematchedlog.class.php (hunk 3 above)
Comment (ddurieux) [reply inc/rulematchedlog.class.php]: Fixed

File: inc/rulematchedlog.class.php (hunk 4 above)
Comment (ddurieux) [reply inc/rulematchedlog.class.php]: Transfered in the css

**Comment by ddurieux** [comment]
Replaced by #2885

**Pull request closed without merging**